lib/gitsh/environment.rb

```diff
@@ -39,7 +40,7 @@ def fetch(key, force_default_git_command = false, &block)
         end
       end
     rescue KeyError
-      raise KeyError, "Variable '#{key}' is not set"
+      raise Gitsh::Error, "Variable '#{key}' is not set"
```

Comment:
Good call. I've called it `Gitsh::UnsetVariableError` instead, because I think that better captures the purpose.